From aada3fe34df1d9cdee9d39c33702817a1eb2751f Mon Sep 17 00:00:00 2001
From: Ong Boon Leong <boon.leong.ong@intel.com>
Date: Wed, 27 May 2020 08:58:35 +0800
Subject: [PATCH 48/77] net: stmmac: make stmmac_xsk_tx_umem() returns NULL if
 XDP is OFF

Make the stmmac_xsk_tx_umem() function more robust and if TxQ is not
XDP TX, we simply return NULL here.

Signed-off-by: Ong Boon Leong <boon.leong.ong@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/stmmac_main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
index c44b48a63f8d..0876e0ca43de 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -1449,6 +1449,8 @@ static struct xdp_umem *stmmac_xsk_tx_umem(struct stmmac_priv *priv, u32 queue)
 
 	if (queue_is_xdp(priv, queue))
 		queue -= priv->plat->num_queue_pairs;
+	else
+		return NULL;
 
 	if (!xdp_on || !test_bit(queue, priv->af_xdp_zc_qps))
 		return NULL;
-- 
2.17.0

